test/stateSpec.js

```diff
@@ -24,6 +24,7 @@ describe('state', function () {
       DD = { parent: D, params: { x: null, y: null, z: null } },
       DDDD = { parent: D, controller: function() {}, template: "hey"},
       E = { params: { i: {} } },
+      F = { params: { a: '', b: false, c: 0, d: undefined }},
```

Comment:
@eddiemonge I can definitely add -1.  Also I added the test here because I could not find a place where the is method on types were tested.  Is there a better place for this test.  And maybe instead of testing by navigating to a state with default params all of the `is` methods should be tested separately.